index.bs

```diff
@@ -508,26 +507,32 @@ To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
     Issue: Nothing seems to define what [=status codes|status code=]
     is used for UTF-8 errors.
 
- 1. Match |data| against the [=remote end definition=]. If this results in a
+1. Let |parsed| be the result of [=parse JSON into Infra values|parsing JSON
+   into Infra values=] given |data|. If this throws an exception, then [=respond
+   with an error=] given |connection|, null, and [=invalid argument=], and
+   finally return.
+
+ 1. Match |parsed| against the [=remote end definition=]. If this results in a
     match:
 
-    1. Let |parsed| be the map representing the matched data.
+    1. Let |matched| be the map representing the matched data.
 
-    1. Assert: |parsed| contains "<code>id</code>", "method", and
-       "<code>params</code>"
+    1. Assert: |matched| contains "<code>id</code>", "<code>method</code>", and
```

Comment:
Historically we have ignored unknown keys for forward-compat.

Reply:
That's precisely what I'm trying to express here, but have apparently failed. My intention was that |matched| effectively be the subset of |parsed| that matches the CDDL.

Reply:
But if the remote end definition has `{*text => text}` then it will match the other fields too. If it doesn't have that then I think it will reject extra fields.

Reply:
So that's what `{*text => any}` means... and you're saying we need that for the definition to match at all? This sounds like a problem, I'll file an issue.

What should I do here for now? A note saying how |matched| is different from |parsed| would be good. But it sounds like actually they would be identical if there was a match, right?

Reply:
Right, I think it would be identical (or maybe not quite if we start using the default value features). You could handwave in some way about matched containing only the explictly named fields. Or just remove "and no others".

Reply:
I've filed https://github.com/w3c/webdriver-bidi/issues/79.

Reply:
FWIW I'm happy to land this if you just delete "and no others" for now.

Reply:
OK, I'll do just that.